lv_conf_template.h

```diff
@@ -333,8 +333,11 @@
 /* Add `id` field to `lv_obj_t` */
 #define LV_USE_OBJ_ID           0
 
+/* Automatically assign an ID when obj is created */
+#define LV_USE_OBJ_ID_ASSIGN    1
```

Comment:
```suggestion
# if LV_USE_OBJ_ID != 0
/* Automatically assign an ID when obj is created. */
#define LV_OBJ_ID_AUTO_ASSIGN    1
```

Reply:
I found out we can use `#define LV_OBJ_ID_AUTO_ASSIGN    LV_USE_OBJ_ID ` So it's automatically enabled and always defined. 

Do you think it's acceptable?

Reply:
The generated conf.h looks like this.
```c
/* Automatically assign an ID when obj is created */
#ifndef LV_OBJ_ID_AUTO_ASSIGN
    #ifdef CONFIG_LV_OBJ_ID_AUTO_ASSIGN
        #define LV_OBJ_ID_AUTO_ASSIGN CONFIG_LV_OBJ_ID_AUTO_ASSIGN
    #else
        #define LV_OBJ_ID_AUTO_ASSIGN   LV_USE_OBJ_ID
    #endif
#endif
```

So LV_OBJ_ID_AUTO_ASSIGN is already defined, and default to 1 if `LV_USE_OBJ_ID` is enabled.

This is what we want(always define the macro with correct value).

Reply:
I understand that you want to use `LV_OBJ_ID_AUTO_ASSIGN` as a sort of compound macro that, when `1`, means `LV_USE_OBJ_ID && (!defined(LV_OBJ_ID_AUTO_ASSIGN) || (defined(LV_OBJ_ID_AUTO_ASSIGN) && LV_OBJ_ID_AUTO_ASSIGN != 0))`.

Until now, using defines like `LV_DEMO_MUSIC_LARGE` when `LV_USE_DEMO_MUSIC=0` is "undefined behavior". Code that uses `LV_DEMO_MUSIC_LARGE` first checks that `LV_USE_DEMO_MUSIC == 1` before considering the value of `LV_DEMO_MUSIC_LARGE`. In your changes, you use `LV_OBJ_ID_AUTO_ASSIGN` as a shortcut.

It makes sense, but I cannot say whether it's okay to break the old pattern for the first time. @kisvegabor?

Reply:
You are correct. This breaks the old pattern.
For system using Kconfig, it normally uses `#ifdef` instead of `#if `. 

LVGL follow the latter style, thus it makes sense to discuss whether we should always define those macros, either enabled as 1 or disabled as 0.

I can follow the old pattern in this PR and raise another discussion if that's more appropriate.

Reply:
@liamHowatt just pointed out this unresolved thread.

If we use `#if` it would really make more sense to **not** add guards around the child configs.  Let's keep it as it is in this feature, and discuss it a new issue.